src/enzo/Grid_LightBosonInitialize.C

```diff
@@ -0,0 +1,248 @@
+    return SUCCESS;
+  }
+
+  int field;
+  for (field = 0; field < NumberOfBaryonFields; field++)
+    if (BaryonField[field] == NULL)
+      BaryonField[field] = new float[size];
+
+
+  float TemperatureUnits = 1, DensityUnits = 1, LengthUnits = 1,
+    VelocityUnits = 1, TimeUnits = 1;
+
+  if (QuantumGetUnits(&DensityUnits, &LengthUnits, &TemperatureUnits,
+         &TimeUnits, &VelocityUnits, Time) == FAIL) {
+    ENZO_FAIL("Error in GetUnits.");
+  }
+
+  int RePsiNum, ImPsiNum, FDMDensNum;
+  RePsiNum = FindField(RePsi, FieldType, NumberOfBaryonFields);
+  ImPsiNum = FindField(ImPsi, FieldType, NumberOfBaryonFields);
+  FDMDensNum = FindField(FDMDensity, FieldType, NumberOfBaryonFields);
+
+
+  float coef = (5.9157166856e27*TimeUnits/pow(LengthUnits,2));
+
+  FLOAT x=0,y=0,z=0;
+  float xv;
+
+  //float a = 1./(1.+2.);
+  float alpha = 1./500;
+  float initialtime=-0.1;
+  float sumsquare = pow(alpha,2)+pow(coef*initialtime,2);
+
+  float a = 0.1;
+  float expa, expb;
+  float pi = 3.1415927;
+  float k1 = 2*pi;
+  float k2 = 2*pi;
+
+  int i,j,k;
+  for (k = 0; k < GridDimension[2]; k++) {
+  for (j = 0; j < GridDimension[1]; j++) {
+  for (i = 0; i < GridDimension[0]; i++) {
+
+    x = CellLeftEdge[0][i] + 0.5*CellWidth[0][i] - CenterPosition;
+    xv = CellLeftEdge[0][i] - CenterPosition;
+
+    index = GRIDINDEX_NOGHOST(i,j,k); 
+
+    // set up an initial Gaussian density profile, v=0
+
+// 1) Gaussian Density Field
+      float r1,i1,phase,r2,i2;
+      phase = -pow(x,2)/sumsquare/2*coef*initialtime;
+      r1 = cos(phase);
+      i1 = -sin(phase);
+      r2 = sqrt((sqrt(sumsquare)+alpha)/2.);
+      i2 = -sqrt((sqrt(sumsquare)-alpha)/2.);
+      BaryonField[iden ][index] = 1*exp(-alpha*pow(x,2)/sumsquare)/pow(sumsquare,0.5)/5; 
+      BaryonField[RePsiNum][index] = 1*exp(-alpha*pow(x,2)/sumsquare/2.)/sqrt(sumsquare)*(r1*r2-i1*i2); //sqrt(BaryonField[iden ][index]);
+      BaryonField[ImPsiNum][index] = 1*exp(-alpha*pow(x,2)/sumsquare/2.)/sqrt(sumsquare)*(r1*i2+i1*r2);
+      BaryonField[FDMDensNum][index] = BaryonField[RePsiNum][index]*BaryonField[RePsiNum][index] + BaryonField[ImPsiNum][index]*BaryonField[ImPsiNum][index];
+
+      //BaryonField[ivx ][index] = 1.e-4*exp(-alpha*pow(x,2)/sumsquare); // Gaussian Density Field
+      BaryonField[ivx  ][index] = xv/sumsquare*coef*coef*initialtime;
+      BaryonField[ietot][index] = 100; 
+// 2) Fresnel solution
+    /*if( x>0){
+      BaryonField[iden ][index] = 6.;
+      BaryonField[RePsiNum][index] = sqrt(6.);
+      BaryonField[ImPsiNum][index] = 0;
+    }else if (x==0.){
+      BaryonField[iden ][index] = 5.5;
+      BaryonField[RePsiNum][index] = sqrt(5.5);
+      BaryonField[ImPsiNum][index] = 0;
+    }else{
+      BaryonField[iden ][index] = 5.;
+      BaryonField[RePsiNum][index] = sqrt(5.);
+      BaryonField[ImPsiNum][index] = 0;
+    }
+
+// 3) Zeldovich Test
+      //BaryonField[iden ][index] = (10.0+5.0*cos(2*3.1415927*(x-0.5)))/10.;
```

Comment:
can you just delete the commented-out lines in this file, or are there compelling reason to keep them?

Reply:
These lines corresponded to four different tests; a runtime parameter has now been added to select between the tests.